[feat-4951]: add subfilters to configuration

vdegraaf opened the pull request:
Ticket: [SIG-4951](https://datapunt.atlassian.net/browse/SIG-4951)

## Context
The values of the maincategories of which the subcatefory filters were enabled was hard coded. Since other municipalities want to have different filters we want to make them configurable.

## Changes
- Added hasSubcategoriesEnabled to the map options in the configuration file
- Adjust tests

## Release notes
- we should add this configuration in the base file in azure config of acceptance and production
- We should set the correct config for Amsterdam in azure config

[SIG-4951]: https://datapunt.atlassian.net/browse/SIG-4951?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

SireeKoolenWijkstra reviewed (changes requested):
When overig is selected, then that category will have a chevrondown etc, but you can't uncheck the main category nor the subcategory of overig. Probably because it doesn't have subcategories of its own. It works like a charm with the other categories

OscarBakker commented on src/signals/IncidentMap/components/FilterPanel/utils.test.ts:
```diff
@@ -34,3 +38,19 @@ describe('getFilterCategoriesWithIcon', () => {
 })
+
+describe('showSubCategoryFilter', () => {
+  it('should return false on default', () => {
```
might wanne add a configuration mock reset here, when adding a new test, you don't want configuration.map.optionsIncidentMap.hasSubfiltersEnabled be set is some other it block.

vdegraaf replied:
good one!

vdegraaf commented:
> When overig is selected, then that category will have a chevrondown etc, but you can't uncheck the main category nor the subcategory of overig. Probably because it doesn't have subcategories of its own. It works like a charm with the other categories

It doesn't work because the name of the main and subcategory are the same which causes the update function to break. I changed it to use slugs because they should be identical. However, the slugs of "overig" are also the same, making it impossible to set subFilters to the "Overig" category. Which might not be an issue since it only has one public accessible sub-filter. I added this to the description in the config file.
